lib/matplotlib/colors.py

```diff
@@ -1225,6 +1240,594 @@ def reversed(self, name=None):
+        rgba = np.asarray(rgba)
+        for c, xx in zip(self[1:], X[1:]):
+            sub_rgba, sub_mask_bad = c(xx, bytes=False, return_mask_bad=True)
+            sub_rgba = np.asarray(sub_rgba)
+            rgba[..., :3] += sub_rgba[..., :3]  # add colors
+            rgba[..., 3] *= sub_rgba[..., 3]  # multiply alpha
+            mask_bad |= sub_mask_bad
+
+        if self.combination_mode == 'Sub':
+            rgba[..., :3] -= len(self) - 1
+
+        rgba[mask_bad] = self.get_bad()
+
+        rgba = np.clip(rgba, 0, 1)
+
+        if alpha is not None:
+            alpha = np.clip(alpha, 0, 1)
+            if alpha.shape not in [(), np.array(X[0]).shape]:
+                raise ValueError(
+                    f"alpha is array-like but its shape {alpha.shape} does "
+                    f"not match that of X[0] {np.array(X[0]).shape}")
+            rgba[..., -1] *= alpha
+
+        if bytes:
+            rgba = (rgba * 255).astype('uint8')
+
+        if not np.iterable(X[0]):
+            rgba = tuple(rgba)
+
+        return rgba
+
+    def copy(self):
+        """Return a copy of the multivarcolormap."""
+        return self.__copy__()
+
+    def __copy__(self):
+        cls = self.__class__
+        cmapobject = cls.__new__(cls)
+        cmapobject.__dict__.update(self.__dict__)
+        cmapobject.colormaps = [cm.copy() for cm in self.colormaps]
+        cmapobject._rgba_bad = np.copy(self._rgba_bad)
+        return cmapobject
+
+    def __eq__(self, other):
+        if not isinstance(other, MultivarColormap):
+            return False
+        if not len(self) == len(other):
+            return False
+        for c0, c1 in zip(self, other):
+            if not c0 == c1:
+                return False
+        if not all(self._rgba_bad == other._rgba_bad):
+            return False
+        if not self.combination_mode == other.combination_mode:
+            return False
+        return True
+
+    def __getitem__(self, item):
+        return self.colormaps[item]
+
+    def __iter__(self):
+        for c in self.colormaps:
+            yield c
+
+    def __len__(self):
+        return len(self.colormaps)
+
+    def __str__(self):
+        return self.name
+
+    def get_bad(self):
+        """Get the color for masked values."""
+        return np.array(self._rgba_bad)
+
+    def set_bad(self, color='k', alpha=None):
+        """Set the color for masked values."""
+        self._rgba_bad = to_rgba(color, alpha)
+
+    @property
+    def combination_mode(self):
+        return self._combination_mode
+
+    @combination_mode.setter
+    def combination_mode(self, mode):
+        if mode not in ['Add', 'Sub']:
+            raise ValueError("Combination_mode must be 'Add' or 'Sub',"
+                             f" {mode!r} is not allowed.")
+        self._combination_mode = mode
+
+    def _repr_png_(self):
+        raise NotImplementedError("no png representation of MultivarColormap"
```

Comment:
I can return them as a tuple or join them into one image.
On the other hand, `_repr_png_()` is a private function, and it is not used by `_repr_html_()`, do you know if it is used by anything else? 
If not it might be more sensible to remove it and only implement `_repr_png_()` after a use case arises.

Reply:
I added a `_repr_png_()`, that returns a single png, but I am not sure how to set up an automated test for it.